monkestation/code/game/objects/items/anime.dm

```diff
@@ -108,15 +87,82 @@
 	M.equip_in_one_of_slots(new_item, slots , qdel_on_fail = TRUE)
 
 /obj/item/choice_beacon/anime/generate_display_names()
-	var/static/list/anime
+	var/list/anime
 	if(!anime)
 		anime = list()
-		var/list/templist = list(/obj/item/anime/cat, //Add to this list if you want your implant to be included in the trait
-								 /obj/item/anime/fox,
-								 /obj/item/anime/wolf,
-								 /obj/item/anime/shark
-								)
+		var/list/templist = list()
+		if(hacked_l == TRUE)
+			templist = list(/obj/item/anime/cat/hacked,
+							 /obj/item/anime/fox/hacked,
+							 /obj/item/anime/wolf/hacked,
+							 /obj/item/anime/shark/hacked
+							)
+		else
+			templist = list(/obj/item/anime/cat, //Add to this list if you want your implant to be included in the trait
+							 /obj/item/anime/fox,
+							 /obj/item/anime/wolf,
+							 /obj/item/anime/shark
+							)
 		for(var/V in templist)
 			var/atom/A = V
 			anime[initial(A.name)] = A
 	return anime
+
+/obj/item/anime/hacked/afterattack(mob/living/carbon/user)
+	if(ishuman(user))
+		anime_transformation(user)
+
+/obj/item/anime/hacked
+	name = "hacked anime dermal implant"
+	hacked = TRUE
+
+/obj/item/anime/cat/hacked
+	name = "hacked anime cat dermal implant"
+	hacked = TRUE
+
+/obj/item/anime/fox/hacked
+	name = "hacked anime fox dermal implant"
+	hacked = TRUE
+
+/obj/item/anime/wolf/hacked
+	name = "hacked anime wolf dermal implant"
+	hacked = TRUE
+
+/obj/item/anime/shark/hacked
+	name = "hacked anime shark dermal implant"
+	hacked = TRUE
+
+/obj/item/choice_beacon/anime/hacked
+	name = "hacked anime dermal implant kit"
+	hacked_l = TRUE
+
+/obj/item/anime/proc/anime_transformation(mob/living/carbon/user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/weeb = user
+		var/new_color = input(user, "Choose an Anime color:", "Anime Color (clicking 'cancel' will set Anime color to Hair color):", weeb.hair_color) as color|null
```

Comment:
This will be a hard stun due to a UI element when it hits a target.
Make it randomize the color if it's hacked.

Reply:
should be fixed by no UI stun.